pkg/snmp/traps/config.go

```diff
@@ -60,12 +82,90 @@ func (c *Config) Addr() string {
 	return fmt.Sprintf("%s:%d", c.BindHost, c.Port)
 }
 
-// BuildV2Params returns a valid GoSNMP SNMPv2 params structure from configuration.
-func (c *Config) BuildV2Params() *gosnmp.GoSNMP {
-	return &gosnmp.GoSNMP{
-		Port:      c.Port,
-		Transport: "udp",
-		Version:   gosnmp.Version2c,
-		Logger:    gosnmp.NewLogger(&trapLogger{}),
+func (c *Config) getAgentHostname() string {
+	hostname, err := util.GetHostname(context.TODO())
+	if err != nil {
+		// this scenario is not likely to happen since
+		// the agent cannot start without a hostname
```

Comment:
Would it be possible to get the agent hostname from initialization code, so we don't have to deal with a potential error here?